src/main/java/com/company/dao/EngineDao.java

```diff
@@ -3,7 +3,18 @@
 import com.company.engine.Engine;
 
 import java.util.List;
+import java.util.Optional;
 
 public interface EngineDao {
 
+    int save(Engine engine);
+
+    int delete(int id);
+
+    int update(Engine engine, int id);
```

Comment:
Why do you need id separately? You can create EngineEntity with all needed fields